examples/helloworld/priv/protos/helloworld.proto

```diff
@@ -25,3 +27,15 @@ message HelloReply {
   string message = 1;
   google.protobuf.Timestamp today = 2;
 }
+
+message HelloSpec {
+  string message = 1;
+  Spec spec = 2;
+
+  message Spec {
```

Comment:
Spec should resolve if it has a package name == `example.helloworld.HelloSpec`, but we give it a package name of `example.helloworld`

Reply:
I believe this issue is due to incorrect `dependency` settings in the reflection server. In this case, changing the nested message's package to the parent's message type fixes the issue by coincidence. Our `DescFile` in the `state` shows the file `example.helloworld.HelloSpec.proto` has a dependency on `["example.helloworld.HelloSpec.Spec"]`, which is incorrect. Removing this dependency field will also fix the issue, but it might cause problems if our message imports other protos. I will investigate this issue further later this week.

If you need to integrate `grpc_reflection` into your services urgently, I think changing the package name as a temporary solution is acceptable to me.

Reply:
> We should omit types from the dependencies list when the descriptor is already included there

Yes, that's what I mean. However, I need to focus on the Istio migration for now. I'll conduct more tests this weekend.